packages/dds/merge-tree/src/opBuilder.ts

```diff
@@ -88,8 +91,31 @@ export function createRemoveRangeOp(start: number, end: number): IMergeTreeRemov
  *
  * @internal
  */
-// eslint-disable-next-line import/no-deprecated
-export function createObliterateRangeOp(start: number, end: number): IMergeTreeObliterateMsg {
+export function createObliterateRangeOp(
+	start: SequencePlace,
+	end: SequencePlace,
+	mergeTreeEnableSidedObliterate: boolean,
+	// eslint-disable-next-line import/no-deprecated
+): IMergeTreeObliterateMsg {
+	if (mergeTreeEnableSidedObliterate) {
+		const startPlace = normalizePlace(start);
+		const endPlace =
+			typeof end === "number"
+				? { pos: end - 1, side: Side.After } // default to inclusive bounds
+				: normalizePlace(end);
+		return {
+			pos1: startPlace.pos,
+			before1: startPlace.side === Side.Before,
```

Comment:
i'm a little worried about modeling it this way, as a client who doesn't understand sideness will have no problem processing this op, it will just do it incorrectly by only processing pos1 and pos2 and ignoring before1 and before2, this will lead to eventually consistency errors which are quick hard to debug

Reply:
that, or allow pos1 and pos2 to be objects